src/LatteContext/Collector/TemplateRenderCollector.php

```diff
@@ -137,7 +138,12 @@ public function processNodeInPhp(Node $node, Scope $scope): ?array
         }
 
         $calledType = $scope->getType($node->var);
-        if (!$this->templateTypeResolver->resolve($calledType)) {
+        if ($calledType instanceof ThisType) {
+            $calledType = $calledType->getStaticObjectType();
+        }
+        $calledType = TypeCombinator::removeNull($calledType);
+        $engineType = new ObjectType('Latte\Engine');
```

Comment:
just question, why we use Latte only here? It could be in TemplateTypeResolver.php with other classes, or not?

Reply:
We use templateTypeResolver in different context that are not valid for Engine

Reply:
You are right LAtte\Runtime\Template acnnot be used this was and calling render on it is marked as internal so I think we can ignore this use-case a vover only Engine.